src/wallet/wallet.cpp

```diff
@@ -964,15 +964,30 @@ bool CWallet::AddToWallet(const CWalletTx& wtxIn, bool fFromLoadWallet, CWalletD
 
 bool CWallet::UpdatedNoteData(const CWalletTx& wtxIn, CWalletTx& wtx)
 {
+    // On reindex, IncrementNoteWitnesses() is called for every block again. So
+    // we drop the cached witnesses for any transactions we already know about.
+    bool dropCache = GetBoolArg("-reindex", false);
```

Comment:
This will be true not just on startup when we are doing the reindexing, but for the remaining life of that zcashd process. Is that okay?